### DISCUSSION TITLE
Version 1 Alpha 1

### DISCUSSION THREAD

**Discussion by stuyam** [bootstrap-email/bootstrap-email#84] [General]
1.0.0.alpha1 was just released. [RubyGem](https://rubygems.org/gems/bootstrap-email) can be found here and new setup guide here [Setup](https://v1.bootstrapemail.com/docs/setup) as well as a [What's New Section](https://v1.bootstrapemail.com/docs/whats-new).

There are new docs for the alpha that are still being polished and adding more content: https://v1.bootstrapemail.com. Any feedback there is also welcome.

There is a new CLI and vanilla ruby implementation that can be used which no longer requires Rails to be the sole way to use Bootstrap Email!

Drop any feedback here on this first alpha release 🙌

**Comment by ComtelJeremy** [comment]
Hi @stuyam, this is great news! We will start trying it out in our pipeline shortly. Will keep you posted.

**Reply by ComtelJeremy** [reply]
Just tried it out quickly... I am getting this:

```
Traceback (most recent call last):
        12: from C:/Ruby27-x64/bin/bootstrap-email:23:in `<main>'
        11: from C:/Ruby27-x64/bin/bootstrap-email:23:in `load'
        10: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/bin/bootstrap-email:2:in `<top (required)>'
         9: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/bin/bootstrap-email:2:in `require_relative'
         8: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap-email/bootstrap_email_cli.rb:1:in `<top (required)>'
         7: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap-email/bootstrap_email_cli.rb:1:in `require_relative'
         6: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap_email.rb:21:in `<top (required)>'
         5: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap_email.rb:21:in `each'         4: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap_email.rb:21:in `block in <top (required)>'
         3: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap_email.rb:21:in `require_relative'
         2: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap-email/components/alert.rb:1:in `<top (required)>'
         1: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap-email/components/alert.rb:2:in `<module:BootstrapEmail>'
C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1/lib/bootstrap-email/components/alert.rb:3:in `<module:Component>': uninitialized constant BootstrapEmail::Component::Base (NameError)
```

Am I missing something? I am not familiar with Ruby, so I could have missed a step.

**Reply by stuyam** [reply]
What is the command you are calling and how did you install it? It may just be a bug or unclear documentation so not your fault thank you for giving it a try so soon! 👍

**Reply by ComtelJeremy** [reply]
```
gem install bootstrap-email --pre
bootstrap-email original.html > complete.html
```

Interestingly I can't call the -h flag either... same error.

**Reply by ComtelJeremy** [reply]
Using ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x64-mingw32] in Command Prompt with Ruby

**Reply by stuyam** [reply]
Great thank you I will take a look!

**Reply by stuyam** [reply]
Just released a new version 1.0.0.alpha1.1, try to reinstall and try again! 🙏  https://rubygems.org/gems/bootstrap-email/versions/1.0.0.alpha1.1

**Reply by ComtelJeremy** [reply]
Progress... Runs, but I get this now (seems like the path is building funny)...
```
Traceback (most recent call last):
        7: from C:/Ruby27-x64/bin/bootstrap-email:23:in `<main>'
        6: from C:/Ruby27-x64/bin/bootstrap-email:23:in `load'
        5: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1.1/bin/bootstrap-email:2:in `<top (required)>'
        4: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1.1/bin/bootstrap-email:2:in `require_relative'
        3: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1.1/lib/bootstrap-email/bootstrap_email_cli.rb:84:in `<top (required)>'
        2: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1.1/lib/bootstrap-email/bootstrap_email_cli.rb:84:in `new'
        1: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1.1/lib/bootstrap-email/compiler.rb:14:in `initialize'
C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/bootstrap-email-1.0.0.alpha1.1/lib/bootstrap-email/compiler.rb:14:in `read': Invalid argument @ rb_sysopen - C:/Users/USER/Downloads/C:/Users/USER/Downloads/SimpleValidationEmail.html (Errno::EINVAL)
```

You'll notice the root path (I am running the CMD window at C:/Users/USER/Downloads) is being repeated?

**Reply by ComtelJeremy** [reply]
I am on Windows 10, if that matters.

**Reply by ComtelJeremy** [reply]
If I run it with no arguments and do the entire folder, it works! Perhaps there is an issue somewhere here: https://github.com/bootstrap-email/bootstrap-email/blob/2ed94191698ea783a11f057a0978e79a44701fa8/lib/bootstrap-email/bootstrap_email_cli.rb#L83 where the path is combined, or something like that?

I am unable to run it specifying a file or path, or output file. I'll try some of the other switches as I have time.

**Reply by stuyam** [reply]
Off to a bumpy start but this is all super valuable thank you. You are right about that line, it was basically doubling up on the working directory path. I added some more tests to also cover more CLI use cases pointed out here so that is great! Just pushed a new release 1.0.0.alpha1.2, want to make sure the core things are working 👍

**Comment by ComtelJeremy** [comment]
Also, a question... for the pattern option, can we specify a relative path for the destination... for example:
emails/* with the destination emails/compiled/* (if you catch my meaning)?
I do think we might need an option to rename the outputted file (as we might want it in the original folder with a new name - for example with a suffix). I imagine we could do this with a Powershell script, or something, or maybe a Ruby script? Just starting to figure out how to work this into a build step.

**Reply by stuyam** [reply]
The path and destination example you described here should already work. Though it would be `bootstrap-email -p 'emails/*' -d 'emails/compiled'` but I did in the latest release at the ability to use the `/*` at the end like `bootstrap-email -p 'emails/*' -d 'emails/compiled/*'` even though it does the same thing.

Question on naming, do you like /output for the default or do you prefer /compiled? I have gone back and forth.

**Comment by ComtelJeremy** [comment]
Just an update for you, @stuyam ...

We have successfully joined your CLI with a bash script in our build step. 🚀 We now compile all our templates and upload them to our storage container when we push our branches in git... works good so far. Therefore, no bootstrap-email-ified &trade; templates in the repo, so we don't have to manage versions, but everything is bootstrap-email-ified &trade; and pushed to our server in a build step.

It's very nice!

We will let you know when it hits production and we have a chance to fully test the templates with live emails (we have some other infrastructure work to do first).

I think we would be willing to share our bash script and some basic details on how we got it running for others who use this library, as time allows.
